posts/git-tips.md

```diff
@@ -0,0 +1,53 @@
+---
+authors:
+- Ben Scott
+tags:
+date: 2021-08-09T11:00:13.000Z
+title: "Git: Dealing with broken branches"
+image: 
+---
+
+Software Engineers are constantly working with Pull Requests(PR). Most organizations have now adopted them as a best practice to ensure code that is merged upstream passes quality checks and peer reviews. I want to share a couple tips around dealing with PRs when issues arise. 
+
+# Resolving Merge Conflicts
+
+If you're in an organization that prefers to rebase their feature branches prior to merging them upstream, and your PR starts to have many commits than you've probably suffered from rebase hell (when conflicts just seem to go on forever). 
+
+For example, let's say you branched off `develop` with your `feature` branch. Your branches have diverged and develop now has 10 new commits, and your feature branch has 100 commits. You've tried rebasing, but you end up in an endless loops of conflicts.
+
+The first thing you should do is enable [reuse recorded resolution](https://git-scm.com/docs/git-rerere) . This will cause git to apply the same resolution when it finds the same conflict. To enable it by default for all your repos:
+
+```$ git config --global rerere.enabled 1```
+
+This can solve your issue, but if it doesn't you may just want to squash your commits before updating from `develop`. You could use [interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) if you prefer to do everything in your terminal, however I have found that the easiest way to squash your branch to keep working on it is to use your git provider's squash feature. 
+
+Here are the steps:
+
+* Find the parent commit of your feature branch
+* Create a new feature branch off that commit `feature-squashing`
+* Create a PR from your `feature` branch to `feature-squashing`
+* Click the merge and squash button
```

Comment:
I'm not sure if you have the time, but a graphic would help illustrate what you are talking about here.